### ISSUE TITLE
Second derivative of scipy.special.gammainc

### ISSUE THREAD

**Issue by sirmurphalot** [google/jax#7922]
I need to take the derivative of a Gamma CDF with respect to the shape parameter _alpha_.   I can do this pretty easily with
```python
from jax import scipy, grad
def gamma_cdf_term(alpha, data_value):
    return scipy.special.gammainc(alpha, data_value)
```
and Jax's `grad` function.  What doesn't work is that this Gamma CDF derivative is part of a density that I feed into TensorFlow Probability's NoUTurnSampler, which requires a second derivative with respect to _alpha_.  Here is where I hit a wall:
```
NotImplementedError: Differentiation rule for 'igamma_grad_a' not implemented
```
which I assume means that this second derivative is not available.  

My questions: is my assumption correct?  If so, I read on these that you base new features off of user feedback, so let this be my vote that the second derivative get implemented :).  If not, is there anyone who can provide guidance on how I might get this second derivative?

**Comment by zhangqiaorjc** [comment]
@mattjj

**Comment by hawkinsp** [comment]
@srvasude perhaps also

**Comment by apaszke** [comment]
Yes looks like we're missing higher-order derivatives for `igamma`. If we need to define more operations to do this then it might be a challenging task, because both `igamma` and its derivative are implemented [on the XLA level](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/client/lib/math.cc#L716).

**Comment by hawkinsp** [comment]
Well, it shouldn't be too hard to implement the derivative if someone figures out the math. We use the HLO built in C++ as a convenience and so we can share it with TensorFlow, but there's no functionality improvement over building the same HLO from Python.

**Comment by sirmurphalot** [comment]
I'm not sure if this is what you need (since there is an integral term), but I have worked out a form for the second derivative.  I think you do the partial swap that I do with [Leibniz's integral rule](https://en.wikipedia.org/wiki/Leibniz_integral_rule), but someone might want to check me.

<img width="624" alt="gammamath" src="https://user-images.githubusercontent.com/19316122/133622075-42e064e3-6cd7-47cc-9c03-238ed187f7f6.png">
